### PR TITLE
Add Visualization

### LINKED ISSUE #34: Visualizing CF explanations

### PR DESCRIPTION
**Pull request by Grahams-Uncle**
This closes #34

### CODE REVIEW

**Review by BirkhoffG — CHANGES_REQUESTED** [review]
I would suggest to reduce the arguments passed, perhaps just passing `evaluate.Explanation` would be a better choice. 

I also think too many arguments in the plotting functions. We should keep these arguments, but we need to provide default arguments so that users will not be overwhelmed when they use this framework.

For example, I would refactor `swarmplot` into
```python
def swarmplot(
    cf_exp: Explanation,
    plot_name: str = "Swarm Plot",  # The name for the plot
    sample_num: int = 100,  # The number of sample for the plot
    seed: int = 42,  # Seed for PRG for random sampling
    separate_class: bool = True, # Whether to separate the data based on the classifier result
):
    ...
```

`X`, `cfs`, `dm`, `pred_fn` are contained in `Explanation`.

You can keep your current code, except making your current function into a private function (e.g., you can just name `swarmplot` as `_swarmplot`), and write a wrapper `swarmplot` to call this private `_swarmplot` function.

**Comment by Grahams-Uncle** [comment]
fix #34

File: nbs/ipynb_path.py [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # fix jupyter notebook path
 import sys
-sys.path.append('..')
\ No newline at end of file
+sys.path.append('../../../Downloads/cfnet-master-2')
```
Comment (BirkhoffG) [inline nbs/ipynb_path.py:3]: Don't change this file!

**Review by BirkhoffG — CHANGES_REQUESTED** [review]

File: cfnet/plotting.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        X, # Factual data
+        cfs, # Counterfactual examples
+        dm: TabularDataModule, # Data module, containing info for columns
+        plot_name: str,  # The name for the plot
+        sample_num: int,  # The number of sample for the plot
+        seed: int,  # Seed for PRG for random sampling
+        separate_class: bool, # Whether to separate the data based on the classifier result
+        pred_fn: Callable[[jnp.DeviceArray], jnp.DeviceArray], # Predict function
+    ):
+
```
[the remark this answers is not in the document]
Comment (Grahams-Uncle) [reply cfnet/plotting.py:21]: Which function needs an explanation?

**Review by BirkhoffG — APPROVED** [review]

**Pull request merged**